src/components/Popup.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Transition, animated } from 'react-spring'
+import { Card } from '@aragon/ui'
+import { noop } from '../utils'
+import springs from '../springs'
+
+export const TRANSITION_SCALE_DIFF = 0.04
+export const TRANSITION_SPRING = { ...springs.swift, precision: 0.1 }
+
+class Popup extends React.Component {
+  static propTypes = {
+    onRequestClose: PropTypes.func,
+    children: PropTypes.node,
+  }
+  static defaultProps = {
+    onRequestClose: noop,
+  }
+
+  state = { previouslyFocusedElement: null }
+
+  _element = React.createRef()
+
+  _closeRequested = false
+
+  componentDidMount() {
+    this.attachEvents()
+
+    this.setState({ previouslyFocusedElement: document.activeElement })
+    this._element.current.focus()
+  }
+
+  componentWillUnmount() {
+    this.removeEvents()
+  }
+
+  attachEvents() {
+    this._element.current.addEventListener('focusout', this.handleFocusout)
```

Comment:
[`EscapeOutside`](https://github.com/aragon/aragon/blob/9a1d432a1058e2b3f230a8d9792708aeed265b45/src/components/EscapeOutside/EscapeOutside.js) is being moved to `aragon-ui` but still I suggest we abstract this functionality to a reusable Component.

Reply:
I agree, we should use the same component everywhere for this. I didn’t use `EscapeOutside` because of the way it works: if the user click on an iframe (which is quite frequent in our case), no event is sent at all and the popup couldn’t be closed. As a bonus, the event is also used to close the popup when the user is focusing out of it using the keyboard.

A solution could be to use an invisible element over everything, but it would require two clicks to activate other elements, and disable any hover state. Using `pointer-events: none` could work if only there was a way to still intercept clicks, but there is nothing like that. Re-emitting the event is also impossible because of the security model of the iframe.

I think we will eventually need two types of “escape outside” components:

- The current one (`EscapeOutside`), blocking everything until `<Escape>` is pressed or if something is clicked outside. This is useful for blocking elements (e.g. Dialog).
- A new type, similar to what is used here, triggering when `<Escape>` is pressed or when the `focus` moves outside of the main element. This is useful for non-blocking elements (e.g. Popover).

The idea is to move the behavior of this component into the Aragon UI / Lorikeet `Popover` component.

Reply:
> I think we will eventually need two types of “escape outside” components:

I like that, we could even use each to compose views, eg:

```jsx
<EscapeOutside>
  <Focusoutside>
    <div>I will hide if the user clicks outside, uses escape and/or I lose focus :) </div>
  </Focusoutside>
</EscapeOutside>
```